libs/utils/utils.js

```diff
@@ -1023,28 +1024,83 @@ export async function loadMartech({
   }
 
   window.targetGlobalSettings = { bodyHidingEnabled: false };
-  loadIms().catch(() => {});
+  loadIms().catch(() => { });
 
   const { default: initMartech } = await import('../martech/martech.js');
   await initMartech({ persEnabled, persManifests, postLCP });
 
   return true;
 }
 
+/**
+ * Checks if the user is signed out based on the server timing and navigation performance.
+ *
+ * @returns {boolean} True if the user is signed out, otherwise false.
+ */
+function isSignedOut() {
+  const serverTiming = window.performance?.getEntriesByType('navigation')?.[0]?.serverTiming?.reduce(
+    (acc, { name, description }) => ({ ...acc, [name]: description }),
+    {},
+  );
+
+  return !Object.keys(serverTiming || {}).length || serverTiming?.sis === '0';
+}
+
+/**
+ * Enables personalization (V2) for the page.
+ *
+ * @returns {boolean} True if personalization is enabled, otherwise false.
+ */
+export function enablePersonalizationV2() {
+  const enablePersV2 = document.head.querySelector('meta[name="personalization-v2"]');
+  return enablePersV2 && isSignedOut();
+}
+
 async function checkForPageMods() {
   const {
     mep: mepParam,
     mepHighlight,
     mepButton,
     martech,
   } = Object.fromEntries(PAGE_URL.searchParams);
+  let targetInteractionPromise = null;
   if (mepParam === 'off') return;
   const pzn = getMepEnablement('personalization');
   const promo = getMepEnablement('manifestnames', PROMO_PARAM);
   const target = martech === 'off' ? false : getMepEnablement('target');
   const xlg = martech === 'off' ? false : getMepEnablement('xlg');
+
   if (!(pzn || target || promo || mepParam
     || mepHighlight || mepButton || mepParam === '' || xlg)) return;
+
+  const enablePersV2 = enablePersonalizationV2();
+  if (martech !== 'off' && (target || xlg || pzn) && enablePersV2) {
+    const params = new URL(window.location.href).searchParams;
+    const calculatedTimeout = parseInt(params.get('target-timeout'), 10)
+  || parseInt(getMetadata('target-timeout'), 10)
+  || TARGET_TIMEOUT_MS;
+
+    const { locale } = getConfig();
+    targetInteractionPromise = new Promise((res) => {
+      import('../martech/helpers.js').then(({ loadAnalyticsAndInteractionData }) => {
+        res(loadAnalyticsAndInteractionData(
```

Comment:
`loadAnalyticsAndInteractionData` is an async function, which means it returns a promise, but since we're doing `resolve(loadAnalyticsAndInteractionData(...args))` the resulting type of `targetInteractionPromise` is not `Promise<T>`, it's `Promise<Promise<T>>` (Correct me if I'm wrong here.
One option would be to await `loadAnalyticsAndInteractionData` inside the `.then`, or you could do:
```javascript
const targetInteractionPromise = (async () => {
  const { loadAnalyticsAndInteractionData } = await import('../martech/helpers.js');
  const data = await loadAnalyticsAndInteractionData(...args);
  return data;
})();
```
So you get the sequencing and scheduling behvior of the `.then` snippet above with the semantics of async and await. And since we aren't `await`ing the IIFE, `targetInteractionPromise` has the correct type of `Promise<T>`.